### DISCUSSION TITLE
An option to cache data retrieved via https

### DISCUSSION THREAD

**Discussion by ericemc3** [duckdb/duckdb#10313] [General]
Could we mobilize a data cache system with DuckDB Cli, as it looks possible with DuckDB-wasm? 
An example to illustrate this. I am querying a 1 Go parquet file (all companies in France), and looking inside Toulouse city:
```
SELECT geo_adresse, denominationUniteLegale, activitePrincipaleEtablissement 
FROM 'https://static.data.gouv.fr/resources/sirene-geolocalise-parquet/20240107-143656/sirene2024-geo.parquet'
WHERE codeCommuneEtablissement = '31555'  -- companies in Toulouse
AND denominationUniteLegale ilike '%idlr%' -- companies WITH idlr in their name
-- 7 Mo (EXPLAIN ANALYZE) 
```

```
SELECT geo_adresse, denominationUniteLegale, activitePrincipaleEtablissement 
FROM 'https://static.data.gouv.fr/resources/sirene-geolocalise-parquet/20240107-143656/sirene2024-geo.parquet'
WHERE codeCommuneEtablissement = '31555'  -- companies in Toulouse
AND denominationUniteLegale ilike '%etincelle%' -- companies WITH etincelle in their name
-- 7 Mo (EXPLAIN ANALYZE) 
```

If i run these queries on https://shell.duckdb.org/,
the first one will consume 12 Mo (as per Chrome network inspector),
but the second 0 Mo, thanks to the browser cache.

`enable_http_metadata_cache `or `enable_object_cache ` config options apparently cache only metadata.

**Comment by redox** [comment]
I randomly stumbled upon this discussion - for readers, it's worth considering https://github.com/dentiny/duck-read-cache-fs/
